src/Microsoft.OData.Edm/Csdl/Serialization/EdmModelCsdlSchemaJsonWriter.cs

```diff
@@ -384,10 +384,18 @@ internal override void WriteEnumMemberElementHeader(IEdmEnumMember member)
         /// 7.2.1 Nullable, A Boolean value specifying whether a value is required for the property.
         /// </summary>
         /// <param name="reference">The Edm type reference.</param>
-        internal override void WriteNullableAttribute(IEdmTypeReference reference)
+        /// <param name="alwaysWrite">Specifies if the attribute is always written or not if the value is the default value.
+        internal override void WriteNullableAttribute(IEdmTypeReference reference, bool alwaysWrite)

```

Comment:
When is the `alwaysWrite` parameter ever set to true and why? what determines whether it should be set true or false?

Reply:
It gets called from EdmModelCsdlSerializationVisitor.ProcessFacets 

And that method tests if the property is of a collection type and if the serializer wants to always serialize the nullable attributes for collection (which is true for XML, false for JSON)